ui/ducks/send/send.js

```diff
@@ -0,0 +1,1442 @@
+      );
+    } else {
+      const { from, to: tokenAddress, gas: gasLimit, gasPrice } = txParams;
+      const tokenAmountInDec = getTokenValueParam(tokenData);
+      const address = getTokenAddressParam(tokenData);
+      const nickname = getAddressBookEntry(state, address)?.name ?? '';
+
+      const tokenAmountInHex = addHexPrefix(
+        conversionUtil(tokenAmountInDec, {
+          fromNumericBase: 'dec',
+          toNumericBase: 'hex',
+        }),
+      );
+
+      await dispatch(
+        updateSendAsset({
+          type: ASSET_TYPES.TOKEN,
+          details: { ...assetDetails, address: tokenAddress },
+        }),
+      );
+
+      await dispatch(
+        actions.editTransaction({
+          id: transactionId,
+          gasLimit,
+          gasPrice,
+          from,
+          amount: tokenAmountInHex,
+          address,
+          nickname,
+        }),
+      );
+    }
+  };
+}
+
+// Selectors
+
+// Gas selectors
+export function getGasLimit(state) {
+  return state[name].gas.gasLimit;
+}
+
+export function getGasPrice(state) {
+  return state[name].gas.gasPrice;
+}
+
+export function getGasTotal(state) {
+  return state[name].gas.gasTotal;
+}
+
+export function gasFeeIsInError(state) {
+  return Boolean(state[name].gas.error);
+}
+
+export function getMinimumGasLimitForSend(state) {
+  return state[name].gas.minimumGasLimit;
+}
+
+export function getGasInputMode(state) {
+  const isMainnet = getIsMainnet(state);
+  const showAdvancedGasFields = getAdvancedInlineGasShown(state);
+  if ((!isMainnet && !process.env.IN_TEST) || showAdvancedGasFields) {
+    return GAS_INPUT_MODES.INLINE;
+  }
+  return state[name].gas.mode;
+}
+
+// Asset Selectors
+
+export function getSendAsset(state) {
+  return state[name].asset;
+}
+
+export function getSendAssetAddress(state) {
+  return getSendAsset(state)?.details?.address;
+}
+
+// Amount Selectors
+export function getSendAmount(state) {
+  return state[name].amount.value;
+}
+
+export function getIsBalanceInsufficient(state) {
+  return state[name].gas.error === INSUFFICIENT_FUNDS_ERROR;
+}
+export function getSendMaxModeState(state) {
+  return state[name].amount.mode === AMOUNT_MODES.MAX;
+}
+
+export function getSendHexData(state) {
+  return state[name].draftTransaction.userInputHexData;
+}
+
+export function sendAmountIsInError(state) {
+  return Boolean(state[name].amount.error);
+}
+
+// Recipient Selectors
+
+export function getSendTo(state) {
+  return state[name].recipient.address;
+}
+
+export function getIsUsingMyAccountForRecipientSearch(state) {
+  return state[name].recipient.mode === RECIPIENT_SEARCH_MODES.MY_ACCOUNTS;
+}
+
+export function getRecipientUserInput(state) {
+  return state[name].recipient.userInput;
+}
+
+export function getRecipient(state) {
+  return state[name].recipient;
+}
+
+// Overall validity and stage selectors
+
+export function getSendErrors(state) {
+  return {
+    gas: state.send.gas.error,
```

Comment:
The `send-gas-row` component is still expecting this key to be `gasFee` (see the `errorType` prop). We should normalize to use one key for this everywhere - either `gas` or `gasFee`.

Reply:
https://github.com/MetaMask/metamask-extension/pull/10965/commits/fe70a1b7393c941b846cacc4c211f5e364d77f02 i went with using gasFee in this selector because it aligns with the current key in messages.json and until we stop using variables to contain translation keys it might end up messy to change it.